framework/include/base/MooseObject.h

```diff
@@ -82,6 +152,9 @@ class MooseObject :
   /// The name of this object, reference to value stored in InputParameters
   const std::string & _name;
 
+  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
+  const bool & _enabled;
```

Comment:
On Fri, Nov 20, 2015 at 9:26 AM, Cody Permann <notifications@github.com>
wrote:

> In framework/include/base/MooseObject.h
> <https://github.com/idaholab/moose/pull/5985#discussion_r45487271>:
>
> > @@ -82,6 +152,9 @@ class MooseObject :
> >    /// The name of this object, reference to value stored in InputParameters
> >    const std::string & _name;
> >
> > +  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
> > +  const bool & _enabled;
>
> Perhaps, but do you trust the user to make this decision?
>
Yes, many of them are quite capable of using this to solve more complex
problems.

> If you disable a UO, Postprocessor, Kernel, BC, Material even an
> AuxKernel. You risk having the user make the mistake of not handling the
> "stale" values that would be received through the coupling. I can see a
> hundred ways for this to go bad but only very few ways for it to work
> properly. This might be a support nightmare.
>
The only way to enable/disable an object based on time or the solve is by
creating a Control object, so if someone starts complaining about weired
behavior with old values and they have a Controls block, we will know what
to look for.

All this being said, I was under the impression from people wanting
controls (@andrsd) that this type of ability is the primary use. If this is
not the case then we can drop this PR and move on to something else.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5985/files#r45487271>.
>